### PR TITLE
Sync with substrate

### PR DESCRIPTION
**Pull request by forgetso**
Update to substrate commit [7409589ab78e7cb508db873dd6384634d667b8f5](https://github.com/paritytech/substrate/commit/7409589ab78e7cb508db873dd6384634d667b8f5)

### CODE REVIEW

File: runtime/Cargo.toml [hunk 1]
```diff
@@ -29,7 +29,7 @@ frame-executive = { git = "https://github.com/paritytech/substrate", package = "
 sp-api = { git = "https://github.com/paritytech/substrate", package = "sp-api", default-features = false }
 sp-block-builder = { git = "https://github.com/paritytech/substrate", package = "sp-block-builder", default-features = false }
 sp-consensus-aura = { git = "https://github.com/paritytech/substrate", package = "sp-consensus-aura", default-features = false }
-sp-core = { git = "https://github.com/paritytech/substrate", package = "sp-core", default-features = false }
+sp-core = { git = "https://github.com/paritytech/substrate", version = "4.1.0-dev", package = "sp-core", default-features = false }
```
Comment (HCastano) [inline runtime/Cargo.toml:32]:
```suggestion
sp-core = { git = "https://github.com/paritytech/substrate", package = "sp-core", default-features = false }
```

File: node/src/service.rs [hunk 2]
```diff
@@ -12,6 +12,7 @@ use sp_consensus::SlotData;
 use sp_consensus_aura::sr25519::AuthorityPair as AuraPair;
 use std::{sync::Arc, time::Duration};
 
+
```
Comment (HCastano) [inline node/src/service.rs:15]:
```suggestion
```

**Review by HCastano — COMMENTED** [review]
@forgetso can you address the errors from the CI?

**Comment by HCastano** [comment]
@forgetso I was more specifically referring to the CLA and `RustFmt` checks the CI does before we can merge a PR

**Comment by cmichi** [comment]
@forgetso Thanks for the PR! To fix the `gitlab-fmt` CI stage you can execute `cargo fmt --all` in your repository folder, it will then format all files appropriately.

What process did you use to synchronize this repo with the latest Substrate?

Can you update the commit hash in the `README.md` as well?

**Comment by forgetso** [comment]
@HCastano @cmichi that's me applied `rustfmt` and changed the commit in the `README`.

My process was to try to build the repo, check the compile errors, and then address them one by one. Is there some automated way of syncing the changes?

File: runtime/src/lib.rs [hunk 3]
```diff
@@ -85,7 +85,6 @@ pub mod opaque {
 		}
 	}
 }
-
 // To learn more about runtime versioning and what each of the following value means:
```
Comment (cmichi) [inline runtime/src/lib.rs:88]:
```suggestion

// To learn more about runtime versioning and what each of the following value means:
```

**Review by cmichi — APPROVED** [review]

**Comment by cmichi** [comment]
>My process was to try to build the repo, check the compile errors, and then address them one by one. Is there some automated way of syncing the changes?

@forgetso The way I usually do it is to look through all commits at https://github.com/paritytech/substrate/tree/master/bin/node-template, which happened since our last sync. There's no automated way as of now.

**Pull request merged**